framework/decode/dx12_dump_resources.cpp

```diff
@@ -2092,8 +2095,38 @@ void Dx12DumpResources::CopyResourceAsyncRead(graphics::dx12::ID3D12FenceComPtr
         }
     }
 
-    // After copying task is done, write the data to disk, and free it to reduce memory use.
-    active_delegate_->DumpResource(copy_resource_data);
+    // Always dump the resource if not dumping only modifiable resources
+    if (!options_.dump_resources_modifiable_state_only)
+    {
+
+        active_delegate_->DumpResource(copy_resource_data);
+    }
+    else
+    {
+
+        // Determine if this resource is modifiable. It is modifiable if any subresource state
+        // has one of the states in modifiableTransitionStates set, or a subresource state is
+        // D3D12_RESOURCE_STATE_COMMON. We treat D3D12_RESOURCE_STATE_COMMON as modifiable because
+        // that state could be considered modifiable based on how the resource is used.
+        auto source_resource_object_info = get_object_info_func_(copy_resource_data->source_resource_id);
+        auto source_resource             = reinterpret_cast<ID3D12Resource*>(source_resource_object_info->object);
+        auto source_resource_extra_info  = GetExtraInfo<D3D12ResourceInfo>(source_resource_object_info);
+        std::vector<graphics::dx12::ResourceStateInfo> res_infos = source_resource_extra_info->resource_state_infos;
+        bool                                           resourceIsModifiable = false;
+        for (auto it = res_infos.begin(); it != res_infos.end() && !resourceIsModifiable; it++)
+        {
+            resourceIsModifiable =
+                ((modifiableTransitionStates && it->states) || it->states == D3D12_RESOURCE_STATE_COMMON);
+        }
+
+        // Dump the resource if is it in modifiable state
+        if (resourceIsModifiable)
+        {
+            active_delegate_->DumpResource(copy_resource_data);
+        }
```

Comment:
Should `D3D12_RESOURCE_STATE_COMMON` also be added in `modifiableTransitionStates`? Or should it stay separately?

Here is to dump the resource if the whole subresources are modifiable. Should it dump the part modifiable subresources?

Reply:
modifiableTransitionStates is a bitmask of the stated that would trigger a dump. Unfortunately, D3D12_RESOURCE_STATE_COMMON is defined as 0, so can't add this value to modifiableTransitionStates. It gets handled as a special case.

Reply:
That's good. Thanks.